subcommands/devices/config_updates.go

```diff
@@ -80,8 +84,38 @@ func loadSotaConfig(device string) *toml.Tree {
 	return tree
 }
 
+func exitIfNotForce() {
+	if !force {
+		os.Exit(1)
+	}
+}
+
+func versionInt(ver string) int {
+	verI, err := strconv.Atoi(ver)
+	if err != nil {
+		panic(fmt.Sprintf("Invalid version: %v. Error: %s", ver, err))
+	}
+	return verI
+}
+
+func appsMatch(targetApps, proposedApps []string) bool {
+	for _, a := range proposedApps {
+		match := false
+		for _, targetApp := range targetApps {
+			if a == targetApp {
+				match = true
+				break
+			}
+		}
+		if !match {
+			return false
+		}
+	}
+	return true
+}
+
 // Validate the inputs: Must be alphanumeric, a dash, underscore, or comma
-func validateUpdateArgs() {
+func validateUpdateArgs(device *client.Device) {
```

Comment:
Maybe it makes sense to do this validation in the OTA lite API implementation so direct calls to it are validated too?

Reply:
That's a really interesting idea. I'll look into this today.